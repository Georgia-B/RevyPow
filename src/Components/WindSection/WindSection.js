import React from 'react';

const WindSection = (props) => {
    if (!props.wind) {
        return null;
    }
    return (
        <div className="weather-section__wind">
            <h3 className="weather-section__direction">{props.windDirection}</h3>
            <h1 className="weather-section__value">{props.wind}</h1>
            <div>winds</div>
        </div>
    );
}

export default WindSection;
