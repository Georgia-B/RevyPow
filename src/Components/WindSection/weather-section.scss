.weather-section {

    &__wind {
        text-align: center;
        margin: 0 20px;
    }

    &__value {
        margin: 0;
        font-size: 30px;
        font-weight: lighter;
    }

    &__direction {
        margin: 0 0 5px 0;
        font-size: 26px;
        line-height: 1;
        font-weight: lighter;
    }
}

@media (max-width: $breakpoint-small) {
    .weather-section__value {
        font-size: 34px;
    }
}