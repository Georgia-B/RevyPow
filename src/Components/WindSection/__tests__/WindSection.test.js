import React from 'react'
import { cleanup, render } from '@testing-library/react'
import WindSection from '../WindSection';

describe('Components | <WindSection /> with data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<WindSection wind="5 km/h" windDirection="SW" />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <WindSection /> with no data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<WindSection />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})