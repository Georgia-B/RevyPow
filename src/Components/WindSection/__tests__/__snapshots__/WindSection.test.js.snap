// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <WindSection /> with data should render correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-section__wind"
  >
    <h3
      class="weather-section__direction"
    >
      SW
    </h3>
    <h1
      class="weather-section__value"
    >
      5 km/h
    </h1>
    <div>
      winds
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components | <WindSection /> with no data should render correctly 1`] = `<DocumentFragment />`;
